secrets/external.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2023 SAP SE
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package secrets
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+func CheckPWCMD(pwCmd string) (string, error) {
+	if pwCmd == "" && os.Getenv("OS_PASSWORD") != "" {
+		return "", fmt.Errorf("no command specified")
+	}
+	// Retrieve user's password from external command.
+	out, err := exec.Command("sh", "-c", pwCmd).Output()
+	if err != nil {
+		return "", fmt.Errorf("could not retrieve user password using: %s", pwCmd)
+	}
```

Comment:
We should surface errors to the user but I am thinking about changing this into fatal because if the command fails, OS_PASSWORD is probably not containing the correct password and we would probably exit pretty early in the application anyway and all of our applications can't continue if they don't have correct credentials.